File: app/src/main/java/com/mxt/anitrend/receiver/SchedulerDelegate.java
package com.mxt.anitrend.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.mxt.anitrend.util.JobSchedulerUtil;

/**
 * Created by max on 2017/03/03.
 * broadcast receiver for scheduling or cancelling jobs
 */

public class SchedulerDelegate extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if(intent.getAction() != null && intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED))
            JobSchedulerUtil.scheduleJob(context);
    }
}

File: app/src/main/java/com/mxt/anitrend/receiver/SchedulerDelegate.kt
package com.mxt.anitrend.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.mxt.anitrend.util.JobSchedulerUtil
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject

/**
 * Created by max on 2017/03/03.
 * broadcast receiver for scheduling or cancelling jobs
 */

class SchedulerDelegate : BroadcastReceiver(), KoinComponent {

    private val scheduler by inject<JobSchedulerUtil>()

    override fun onReceive(context: Context, intent: Intent?) {
        when (intent?.action) {
            Intent.ACTION_BOOT_COMPLETED ->
                scheduler.scheduleJob()
        }
    }
}